include/zephyr/bluetooth/conn.h

```diff
@@ -2663,6 +2673,23 @@ struct bt_conn_auth_cb {
 	 */
 	void (*pincode_entry)(struct bt_conn *conn, bool highsec);
 #endif
+
+#if defined(CONFIG_BT_APP_PASSKEY)
+	/** @brief Allow the application to provide a passkey for pairing.
+	 *
+	 *  If implemented, this callback allows the application to provide passkeys for pairing.
+	 *  The valid range of passkeys is 0 - 999999. The application shall return the passkey for
+	 *  pairing, or BT_PASSKEY_RAND to generate a random passkey.
+	 *
```

Comment:
The presence (not-NULL-ness) of this callback has an effect. We need to document that.

Reply:
I'm not sure I understand what you mean. Do you mean that the application is responsible for pairing keys when this is implemented? If so, the note below mentions this. Otherwise, please add a suggestion!

Reply:
Resolved. Effect is technically documented. We agreed accessibility of documentation does not need to be prioritized for this API, so we don't need to give further guidance in the documentation.